Stdlib/Data/List/Base.juvix

```diff
@@ -18,12 +18,8 @@ isElement {A} (eq : A -> A -> Bool) (elem : A) : (list : List A) -> Bool
 --- 𝒪(𝓃). Returns the leftmost element of the list satisfying the predicate or
 --- nothing if there is no such element.
 {-# specialize: [1] #-}
-find {A} (predicate : A -> Bool) : (list : List A) -> Maybe A
-  | nil := nothing
-  | (x :: xs) :=
-    if
-      | predicate x := just x
-      | else := find predicate xs;
+find {A} (predicate : A -> Bool) : (list : List A) -> Maybe A :=
+  filter predicate >> headMaybe;
```

Comment:
in a strict language such as juvix this is less efficient than before, because we always traverse the whole list, even if the first element satisfies the predicate